doc/decisions/multiple_file_backends.md

```diff
@@ -0,0 +1,32 @@
+# Multiple File Backends
+
+## Problem
+
+- For XDG in the `system` namespace may contain several files (XDG_CONFIG_DIRS)
+- Unreachable sources (e.g. `curl`) need some fallback
+- As fallback source if some data cannot be stored in some format
+
+## Constraints
+
+## Assumptions
+
+## Considered Alternatives
+
+## Decision
+
+Multiple File Backends are not supported as they:
+
+- make it impossible for admins to modify content of all files
+- do not work atomically (a journal would be needed)
+- do not work together with mmap
+
+Instead all but the first of the files in `XDG_CONFIG_DIRS` are ignored
```

Comment:
As it is specified like this in XDG it would be weird to warn about it. I rather hope that XDG gets more friendly for configuration management tools.

Reply:
> As it is specified like this in XDG it would be weird to warn about it.

Why? It notifies the admin/user about an incompatibility between Elektra and XDG. We could easily add a `system:/elektra` key to disable the warning.

Reply:
It is not so clear if it is an incompatibility or it is "working as intended". As long as `kdb file` returns the correct file, it should be okay from the user experience.

Reply:
I may have misunderstood. Do we only use the first (existing) directory listed in `XDG_CONFIG_DIRS` or do we take first config file we find in one of the directories listed in `XDG_CONFIG_DIRS`?

The second one would be working as intended to me. The first one would be unexpected and I would like a warning about this behaviour.

Reply:
We take the first config file, see: https://github.com/ElektraInitiative/libelektra/blob/6d23610802c772957732826cdc87b1d4e4aeb59b/src/plugins/resolver/filename.c#L396-L401
and it is even documented.

Reply:
Then I would say that is "working as intended", sorry for the misunderstanding.